resources/liquibase/db.changelog.xml

```diff
@@ -0,0 +1,131 @@
+            </column>
+            <column name="title" type="VARCHAR(100)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="isbn" type="VARCHAR(10)">
+                <constraints nullable="false" unique="true" uniqueConstraintName="books_isbn_key"/>
+            </column>
+            <column name="author" type="VARCHAR(50)">
+                <constraints nullable="false"/>
+            </column>
+            <column defaultValueBoolean="true" name="isAvailable" type="BOOLEAN">
+                <constraints nullable="false"/>
+            </column>
+            <column name="createdAt" type="TIMESTAMP WITH TIME ZONE">
+                <constraints nullable="false"/>
+            </column>
+            <column name="updatedAt" type="TIMESTAMP WITH TIME ZONE">
+                <constraints nullable="false"/>
+            </column>
+             <column name="createdBy" type="VARCHAR(50)">
+                <constraints nullable="false"/>
+            </column>
+             <column name="updatedBy" type="VARCHAR(50)">
+                <constraints nullable="false"/>
+            </column>
+             <column name="totalNumber" type="INTEGER">
+                <constraints nullable="false"/>
+            </column>
+             <column name="nosAvailable" type="INTEGER">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+    </changeSet>
+
+    <changeSet author="moroti" id="202302120924">  
+        <dropColumn   
+                columnName="createdBy"  
+                schemaName="public"  
+                tableName="books"/>   
+        <dropColumn 
+            columnName="updatedBy" 
+            schemaName="public"
+            tableName="books"/>  						
+    </changeSet>
+    
+    <changeSet id="202302120936" author="moroti">
+      <addColumn 
+          schemaName="public"
+          tableName="books">
+          <column name="createdBy" type="int"/>
+          <column name="updatedBy" type="int"/>
+      </addColumn>
+    </changeSet>
+
+    <changeSet id="202302121008" author="moroti">
+      <addForeignKeyConstraint baseColumnNames="createdBy"
+          baseTableName="books"
+          constraintName="fk_users_created_by"
+          onDelete="CASCADE"
+          onUpdate="RESTRICT"
+          referencedColumnNames="id"
+          referencedTableName="users"/>
+        <addForeignKeyConstraint baseColumnNames="updatedBy"
+            baseTableName="books"
+            constraintName="fk_users_updatedBy"
+            onDelete="CASCADE"
+            onUpdate="RESTRICT"
+            referencedColumnNames="id"
+            referencedTableName="users"/>
+        <addNotNullConstraint 
+            columnDataType="int"
+            columnName="createdBy"
+            schemaName="public"
+            tableName="books"/>
+        <addNotNullConstraint 
```

Comment:
Nice work by adding the foreign key constraint. However, you don't need to create changeset for the foreign key and constraint separately. You can add both in your column definition. Look this up.

See example

`<column name="account_key_id">
      <constraints nullable="false" uniqueConstraintName="UQ__account_key" unique="true"
                             foreignKeyName="Fk__account_information__account_key" referencedColumnNames="id" referencedTableName="account_key" />
</column>`